### PR TITLE
Warning Alerts, Visualization of current temperature, Bugfixes for night mode and animations

### PR DESCRIPTION
**Pull request by mariokorte**
- Warning Alerts if brightness or custom color is set too low. Manual override possible.
- Added visualization of current color temperature with second transparent slider thumb
- Bugfix for enabling of night mode and possible simultaneous animations

### CODE REVIEW

**Comment by mariokorte** [comment]
I see. Will do that.

**Comment by mariokorte** [comment]
Should be compatible with iOS7 now. Please check that, because I do not own a device with iOs7 anymore ;)

**Comment by mariokorte** [comment]
I tested the UIAlertView code on iOS9. Works as intended. So I guess this should be OK on iOS7 as well.

**Comment by anthonya1999** [comment]
@mariokorte Could you actually make it all just a UIAlertView, only because it will work with all versions?

**Comment by mariokorte** [comment]
I surely could, but I feel this way is cleaner. Also I would like to use a destructive button for the expert option. This is not possible with UIAlertView.

**Comment by mariokorte** [comment]
Please pull this quite soon. There are some bugs that need to be squashed.

**Comment by anthonya1999** [comment]
@mariokorte Sure, will merge this later this evening...thanks for your work on this!

File: GoodNight/BrightnessViewController.m [hunk 1]
```diff
@@ -53,14 +55,49 @@ - (IBAction)brightnessSwitchChanged {
 }
 
 - (IBAction)dimSliderLevelChanged {
+    if (self.dimSlider.value < 0.1f && !warningIgnored){
+        if (![self presentedViewController]){
+            NSString *title = @"Warning";
+            NSString *message = @"If you further reduce the brightness your screen will go completely dark!";
+            NSString *cancelButton = @"Understood";
```
Comment (RSully) [inline GoodNight/BrightnessViewController.m]: I think this could get confusing, it sounds like an acknowledgement

**Comment by RSully** [comment]
@mariokorte also since updating to master (not including this pull request) I've been getting alerts "Time to enable (null)!"

File: GoodNight/BrightnessViewController.m (hunk 1 above)
Comment (mariokorte) [inline GoodNight/BrightnessViewController.m]: Hm, I could use another wording instead of 'understood'. Is 'OK' any better? Or 'Cancel'?

**Comment by mariokorte** [comment]
Yes, strangely the bundle name seems to be nil.
I am thinking of removing the notifications completely as they do not help anything at all. The firing of the notifications is not recognised by the app when it is not in foreground, so it cannot do anything when it's not running. This is like apple intended it to be.
So we don't gain anything from showing them - the screen is turned on by the 'WakeupScreenIfNeeded' efficiently enough.
What do you guys think?

File: GoodNight/BrightnessViewController.m (hunk 1 above)
Comment (mariokorte) [inline GoodNight/BrightnessViewController.m]: P.S.: is it really so ambiguous? Did you run the code and look at the alert? I deliberately did use the destructive button for the option 'I know what I'm doing', so it is displayed in red and can be seen as the dangerous answer in this context.

**Comment by mariokorte** [comment]
@RSully P.S.: The nil-bundleName should've already been fixed in commit 47f5f40ca47a60000cac189b7397cb13284d55f2 by @anthonya1999. Please use the latest source version.
I hope these commits will be pulled soon too, as there are some irritating bugs and glitches fixed here as well.

File: GoodNight/BrightnessViewController.m (hunk 1 above)
Comment (RSully) [inline GoodNight/BrightnessViewController.m]: Cancel would be best, IMO. Really not a big deal though.

**Comment by anthonya1999** [comment]
@mariokorte I'll go ahead and merge this and do some minor cleanup afterwards. Thanks!

**Comment by RSully** [comment]
@mariokorte I'd be in favor of the alerts being optional, at least, if they're not used for anything. I thought they were required (I read somewhere that the only way to change the temperature was for the display to be awake).

**Comment by mariokorte** [comment]
You are right, the display has to be enabled when changing the gammatable, but this is already achieved by aforementioned function 'WakeupScreenIfNeeded'. The notifications are never exactly fired when the change occurs due to the fact that I previously stated. So they are just for informational use.

But as you can disable notifications app wise I will let everything as it is right now, unless anybody feels different about this.

**Pull request merged**